### PR TITLE
Generate the SwiftSyntax docc index using CodeGeneration

### PR DESCRIPTION
**Pull request by ahoppen**
It turns out we need this index because docc generates the left sidebar that contains from it.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by CodaFi — APPROVED** [review]

File: CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift [hunk 1]
```diff
@@ -55,49 +72,51 @@ struct GenerateSwiftSyntax: ParsableCommand {
...
+        GeneratedFileSpec([swiftSyntaxBuilderDir, "generated", "BuildableNodes.swift"], buildableNodesFile),
+        GeneratedFileSpec([swiftSyntaxBuilderDir, "generated", "ResultBuilders.swift"], resultBuildersFile),
+        GeneratedFileSpec([swiftSyntaxBuilderDir, "generated", "SyntaxExpressibleByStringInterpolationConformances.swift"], syntaxExpressibleByStringInterpolationConformancesFile),
       ]
       + BASE_KIND_FILES.map { baseKind in
-        TemplateSpec(sourceFile: syntaxNode(emitKind: baseKind.key), module: swiftSyntaxDir, filename: "syntaxNodes/\(baseKind.value)")
-      }
+        GeneratedFileSpec([swiftSyntaxDir, "generated", "syntaxNodes", baseKind.value], syntaxNode(emitKind: baseKind.key))
+      } + [
+        GeneratedFileSpec([swiftSyntaxDir, "Documentation.docc", "generated", "SwiftSyntax.md"], swiftSyntaxDoccIndex)
```
Comment (kimdv) [inline CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift:116]: Nit pick and don't need to be part of this PR, personally I would have a `private let generatedPath: String = "generated"` next to  `swiftSyntaxDir`, `swiftSyntaxBuilderDir` and to so on.

**Review by kimdv — APPROVED** [review]

**Comment by kimdv** [comment]
And does DocC not support this kind of thing out of the box?

File: CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift (hunk 1 above)
Comment (ahoppen) [reply CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift:116]: Good suggestion. I updated it.

**Comment by ahoppen** [comment]
> And does DocC not support this kind of thing out of the box?

I’m not sure but in either case it’s good if we can sort the types into groups ourselves rather than listing them all together, which is, what I suspect docc would do by default.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by kimdv** [comment]
@swift-ci please test macOS

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Pull request merged**